File: server/src/main/java/cn/nukkit/server/block/BlockCarrot.java
package cn.nukkit.server.block;

import cn.nukkit.server.item.Item;
import cn.nukkit.server.item.ItemCarrot;

import java.util.Random;

/**
 * @author Nukkit Project Team
 */
public class BlockCarrot extends BlockCrops {

    public BlockCarrot(int meta) {
        super(meta);
    }

    public BlockCarrot() {
        this(0);
    }

    @Override
    public String getName() {
        return "Carrot BlockType";
    }

    @Override
    public int getId() {
        return CARROT_BLOCK;
    }

    @Override
    public Item[] getDrops(Item item) {
        if (meta >= 0x07) {
            return new Item[]{
                    new ItemCarrot(0, new Random().nextInt(3) + 1)
            };
        }
        return new Item[]{
                new ItemCarrot()
        };
    }

    @Override
    public Item toItem() {
        return new ItemCarrot();
    }
}

File: src/main/java/cn/nukkit/block/BlockCarrot.kt
package cn.nukkit.block

import cn.nukkit.item.Item
import cn.nukkit.item.ItemCarrot
import cn.nukkit.level.generator
import java.util.*

/**
 * @author Nukkit Project Team
 */
class BlockCarrot @JvmOverloads constructor(meta: Int = 0) : BlockCrops(meta) {
	override val name: String
		get() = "Carrot Block"

	override val id: Int
		get() = BlockID.Companion.CARROT_BLOCK

	override fun getDrops(item: Item): Array<Item?> {
		return if (damage >= 0x07) {
			arrayOf(
					ItemCarrot(0, Random().nextInt(3) + 1)
			)
		} else arrayOf(
				ItemCarrot()
		)
	}

	override fun toItem(): Item? {
		return ItemCarrot()
	}
}